test/python3/libdnf5/rpm/test_package_query.py

```diff
@@ -103,3 +103,20 @@ def test_filter_name(self):
         self.assertEqual(query.size(), 2)
         # TODO(dmach): implement __str__()
         self.assertEqual([i.get_nevra() for i in query], ["pkg-1.2-3.x86_64", "pkg-libs-1:1.3-4.x86_64"])
+
+    def test_resolve_pkg_spec(self):
+        # Test using resolve_pkg_spec() method
+        query = libdnf5.rpm.PackageQuery(self.base)
+        self.assertGreater(query.size(), 0)
+
+        # Test passing empty forms
+        empty_settings = libdnf5.base.ResolveSpecSettings()
+        match, _ = query.resolve_pkg_spec("pkg", empty_settings, True)
+        self.assertTrue(match)
+
+        # Test passing an explicit list of forms
+        name_arch_settings = libdnf5.base.ResolveSpecSettings()
+        name_arch_settings.nevra_forms.append(libdnf5.rpm.Nevra.Form_NA)
+
+        match, _ = query.resolve_pkg_spec("pkg.x86_64", name_arch_settings, True)
```

Comment:
Please can you also test nevra object (`_`) that is returned by `resolve_pkg_spec`?
And what about to test whether query has requested content after `resolve_pkg_spec`?

Reply:
Thanks for the feedback. I will improve the tests.